Create Virtual_file_system.c

Suruchibharti opened the pull request:
complete implementation of a Virtual File System (VFS) in C language.

Features included:
-> In-memory simulation of file storage using circular and doubly linked lists.
->Supports all basic filesystem operations:
-> mkdir, rmdir, create, delete, write, read, ls, cd, pwd, df, exit
->Implements free block management using a doubly linked list.
->Each directory maintains a circular linked list of child nodes (files/subdirectories).
->Uses meaningful variable names in camelCase style for clarity and maintainability.
->Handles all edge cases: duplicate names, missing files, non-empty directories, full disk, etc.
->Displays realistic disk usage statistics (used/free blocks).
->Simulates real file system behavior such as parent traversal and memory cleanup on exit.

Suruchibharti committed:
Create Virtual_file_system.c

complete implementation of a Virtual File System (VFS) in C language.

Features included:
-> In-memory simulation of file storage using circular and doubly linked lists.
->Supports all basic filesystem operations:
-> mkdir, rmdir, create, delete, write, read, ls, cd, pwd, df, exit
->Implements free block management using a doubly linked list.
->Each directory maintains a circular linked list of child nodes (files/subdirectories).
->Uses meaningful variable names in camelCase style for clarity and maintainability.
->Handles all edge cases: duplicate names, missing files, non-empty directories, full disk, etc.
->Displays realistic disk usage statistics (used/free blocks).
->Simulates real file system behavior such as parent traversal and memory cleanup on exit.

Suryanshsrivastv commented:
the program upon exiting is not freeing the allocated memory.

Suruchibharti committed:
Update Virtual_file_system.c

freeing the allocated memory.

Suruchibharti commented:
> freeing the allocated memory.

Hii @Suryanshsrivastv , I made the respective changes kindly review it again.

Suruchibharti committed:
Update Virtual_file_system.c

Improved file and directory name validation in Compact VFS for safer memory handling

Suryanshsrivastv commented:
your code is not parsing the input properly Compact VFS - ready. Type 'exit' to quit.
/ > create testfile
File 'testfile' created successfully.
/ > write testfile "AAAA"
Data written successfully (size=5 bytes).
/ > write testfile "B"
Data written successfully (size=2 bytes).
/ > read testfile
AAAA"B" ,

and , 

 / > create testfile
File 'testfile' created successfully.
/ > write testfile AAAA
Data written successfully (size=3 bytes).
/ > write testfile "B"
Data written successfully (size=2 bytes).
/ > read testfile
AAAB"
/ > exit
Releasing memory...
Memory released. Exiting program.

Suruchibharti committed:
Update Virtual_file_system.c

Fixed input parsing issue — processCommandLine() now correctly handles quoted and unquoted arguments for the write command.

Suruchibharti commented:
> your code is not parsing the input properly Compact VFS - ready. Type 'exit' to quit. / > create testfile File 'testfile' created successfully. / > write testfile "AAAA" Data written successfully (size=5 bytes). / > write testfile "B" Data written successfully (size=2 bytes). / > read testfile AAAA"B" ,
> 
> and ,
> 
> / > create testfile File 'testfile' created successfully. / > write testfile AAAA Data written successfully (size=3 bytes). / > write testfile "B" Data written successfully (size=2 bytes). / > read testfile AAAB" / > exit Releasing memory... Memory released. Exiting program.

I have fixed the input parsing issue — processCommandLine() now correctly handles quoted and unquoted arguments for the write command.
